### PR TITLE
Demo-Spatial Transformations

### LINKED ISSUE #130: Cropping using Images
After searching how to crop an image in the docs, I realized that `PaddedView` can be used to do that. 
Perhaps it would be useful to mention in its docstring that PaddedView also works if the output image size is smaller than the input one, effectively implementing a cropping functionality, so that the function would show up when searching for related keywords.

### PR DESCRIPTION
**Pull request by ashwani-rathee**
Demo for cropping,resizing and rescaling has been added with this and address issue #130 

closes #130

### CODE REVIEW

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# Region of Interest:(y1,y2) sets the range for y-axis and (x1,x2) sets the range for 
+# x-axis of source image.
+
+img_cropped = img_source[ : ,floor(Int, 1/8*img_size[2]) : floor(Int, 7/8*img_size[2])]
+
+# Let's see img_cropped size:
+
+size(img_cropped)
+
+# Ther is another method to do this and that is through using PaddedView,which is shown [here](https://juliaimages.org/stable/democards/examples/spatial_transformation/alpha_compositing/#Alpha-Compositing/)
```
[the remark this answers is not in the document]
Comment (johnnychen94) [reply docs/examples/spatial_transformation/SpatialTransformations.jl:44]: Another thing I'd like to see here is the view version.

```julia
img_cropped = @view img_source[:, floor(Int, 1/8*img_size[2]) : floor(Int, 7/8*img_size[2])]
```

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+# ## Resizing Operation
+
+# Resizing is a method to resize an image to given specific outout image shape.It is different from rescaling as in rescaling we use scaling factor to manipulate image.
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:48]:
```suggestion
# Resizing is a method to resize an image to a given specific output image shape. It is different from rescaling as in rescaling we use a scaling factor to manipulate the image.
```

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+img_square = imresize(img_source,(400,400));
+img_small = imresize(img_source, ratio=1/4);
+img_medium = imresize(img_small, size(img_small).*2);
+mosaicview(img_source, img_square, img_small, img_medium;nrow=1)
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:53]:
```suggestion
mosaicview(img_source, img_square, img_small, img_medium; nrow=1)
```

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+size(img_cropped)
+
+# Ther is another method to do this and that is through using PaddedView,which is shown [here](https://juliaimages.org/stable/democards/examples/spatial_transformation/alpha_compositing/#Alpha-Compositing/)
+
+# ## Resizing Operation
+
+# Resizing is a method to resize an image to given specific outout image shape.It is different from rescaling as in rescaling we use scaling factor to manipulate image.
+
+img_square = imresize(img_source,(400,400));
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:50]:
```suggestion
img_square = imresize(img_source, (400, 400));
```

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+img_small = imresize(img_source, ratio=1/4);
+img_medium = imresize(img_small, size(img_small).*2);
+mosaicview(img_source, img_square, img_small, img_medium;nrow=1)
+
+# ## Rescaling
+
+# Rescale operation resizes an image by a given scaling factor. The scaling factor can 
+# either be a single floating point value, or multiple values - one along each axis. 
+# Image scaling is the process of changing the size of an image and saving the original 
+# proportions.
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:60]:
```suggestion
# Image scaling is the process of changing the size of an image while preserving the original 
# aspect ratio.
```

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+# ## Rescaling
+
+# Rescale operation resizes an image by a given scaling factor. The scaling factor can 
+# either be a single floating point value, or multiple values - one along each axis. 
+# Image scaling is the process of changing the size of an image and saving the original 
+# proportions. 
+
+# ### Rescaling by percentage
+
+percentage_scale=0.6
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:64]:
```suggestion
percentage_scale = 0.6
```

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+new_size = trunc.(Int, size(img_source) .* percentage_scale)
+img_rescaled = imresize(img_source, new_size);
+mosaicview(img_source, img_rescaled; nrow = 1)
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:67]: Normally, we don't add spaces between keyword

```suggestion
mosaicview(img_source, img_rescaled; nrow=1)
```

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+# We calculated new size by estimating the size of frame by multiplying size by scale and
+# then truncated it to Int format.
+
+# ### Rescaling to a specific dimension
+
+new_width = 200
+percentage_scale = new_width / size(img_source)[2];
+new_size = trunc.(Int, size(img_source) .* percentage_scale);
+img_rescaled = imresize(img_source, new_size);
+mosaicview(img_source, img_rescaled; nrow = 1)
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:78]:
```suggestion
mosaicview(img_source, img_rescaled; nrow=1)
```

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+# We calculated new size by estimating the size of frame by multiplying size by scale and
+# then truncated it to Int format.
+
+# ### Rescaling to a specific dimension
+
+new_width = 200
+percentage_scale = new_width / size(img_source)[2];
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:75]:
```suggestion
percentage_scale = new_width / size(img_source, 2);
```

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+new_size = trunc.(Int, size(img_source) .* percentage_scale);
+img_rescaled = imresize(img_source, new_size);
+mosaicview(img_source, img_rescaled; nrow = 1)
+
+# We have updated our scale by percentage solution to calculate scale-percentage 
+# dynamically based on a change in one of the dimensions.
+# Remember: size(sourceimage)[1] corresponds to height, while size(sourceimage)[2] 
+# corresponds to width.
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:83]:
```suggestion
# Remember: `size(sourceimage) == (height, width)`
```

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+# ### Rescaling by two-fold using restrict function
+
+rescaled_both = restrict(img_source); # both side
+rescaled_height = restrict(img_source, 1); # height
+rescaled_width = restrict(img_source, 2); # width
+mosaicview(img_source, rescaled_both, rescaled_height, rescaled_width; nrow = 1)
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:90]:
```suggestion
mosaicview(img_source, rescaled_both, rescaled_height, rescaled_width; nrow=1)
```

**Review by johnnychen94 — COMMENTED** [review]
Even though these are basic operations, it is a very nice explanation and I'm impressed!

I've left tons of format and language suggestions, hope you don't mind.

Two other things I also wish to see in this demo are:

- image rotation: `imrotate`
- image flip: `reverse`. Note that there are two flip options: flip the image data or flip the image axes.

File: docs/examples/spatial_transformation/SpatialTransformations.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashwani-rathee) [reply docs/examples/spatial_transformation/SpatialTransformations.jl:46]: I am playing around with padded view for image pyramids demo,I'll add a example of this here in a day or two.

**Comment by ashwani-rathee** [comment]
Can I make another demo for other transformations?? This has three operations which seems just the right amount for one I think.

**Comment by johnnychen94** [comment]
Of course.

And feel free to resolve the comments I added.

**Comment by johnnychen94** [comment]
Is this PR done?

**Comment by ashwani-rathee** [comment]
I think so,I will change the image simple view to @view version and add padded view  example .But how do I convert padded view array to image??

**Comment by johnnychen94** [comment]
> But how do I convert padded view array to image??

Can you explain this more? I'm not sure if I understand your question clearly.

You don't need to convert a PaddedView array to image, because in JuliaImages, every colorant array is a valid image, i.e., `AbstractArray{<:Colorant}`.

**Comment by ashwani-rathee** [comment]
In pluto,when we use PaddedView the array doesn't directly convert to image or maybe I am missing something

**Comment by johnnychen94** [comment]
This is probably a Pluto issue; If I get it correctly, it implements a different image display pipeline and not using ImageShow:

In IJulia, it is displayed as an image:

image

File: docs/examples/spatial_transformation/SpatialTransformations.jl [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+# title: Cropping, Resizing and Rescaling
+# cover: assets/lighthouse.png
+# author: Ashwani Rathee
+# date: 2020-11-24
+# ---
+
+# This demonstration shows how to use cropping,resizing and rescaling operations on an 
+# image in Julia using ImageTransformations.jl
+
+using ImageCore, ImageShow, ImageTransformations, TestImages, OffsetArrays, ImageShow
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/SpatialTransformations.jl:11]: The duplicated `ImageShow` here.

Might be more concise to just import `Images` if we have a lot of packages to import

```suggestion
using Images, TestImages, OffsetArrays
```

**Review by johnnychen94 — APPROVED** [review]

**Comment by johnnychen94** [comment]
Welcome as a contributor to JuliaImages! @ashwani-rathee

**Comment by ashwani-rathee** [comment]
:)

**Pull request merged**